File: android/app/src/main/java/net/melisma/selko/data/model/CalendarWorkItem.kt
package net.melisma.selko.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.time.Instant

@Serializable
enum class CalendarWorkAction {
    @SerialName("upsert") UPSERT,
    @SerialName("cancel") CANCEL
}

@Serializable
enum class CalendarWorkStatus {
    @SerialName("pending") PENDING,
    @SerialName("processing") PROCESSING,
    @SerialName("succeeded") SUCCEEDED,
    @SerialName("failed") FAILED,
    @SerialName("blocked") BLOCKED,
    @SerialName("superseded") SUPERSEDED
}

@Serializable
data class CalendarWorkItem(
    val id: String,
    @SerialName("event_id") val eventId: String,
    @SerialName("user_id") val userId: String,
    val action: CalendarWorkAction,
    val generation: Long,
    val status: CalendarWorkStatus,
    @SerialName("provider_event_id") val providerEventId: String? = null,
    @SerialName("expected_provider_revision") val expectedProviderRevision: String? = null,
    @SerialName("force_overwrite") val forceOverwrite: Boolean = false,
    val attempts: Int = 0,
    @SerialName("max_attempts") val maxAttempts: Int = 3,
    @SerialName("next_retry_at") val nextRetryAt: Instant? = null,
    @SerialName("failure_code") val failureCode: String? = null,
    @SerialName("failure_detail") val failureDetail: String? = null,
    @SerialName("created_at") val createdAt: Instant? = null,
    @SerialName("updated_at") val updatedAt: Instant? = null,
    @SerialName("completed_at") val completedAt: Instant? = null
)

File: ios/Selko/Features/Events/Models/CalendarWorkItem.swift
import Foundation

enum CalendarWorkAction: String, Codable, Sendable { case upsert, cancel }

enum CalendarWorkStatus: String, Codable, Sendable {
    case pending, processing, succeeded, failed, blocked, superseded
}

struct CalendarWorkItem: Identifiable, Codable, Sendable, Equatable {
    let id: UUID
    let eventId: UUID
    let userId: UUID
    let action: CalendarWorkAction
    let generation: Int
    let status: CalendarWorkStatus
    let providerEventId: String?
    let expectedProviderRevision: String?
    let forceOverwrite: Bool
    let attempts: Int
    let maxAttempts: Int
    let nextRetryAt: Date?
    let failureCode: String?
    let failureDetail: String?
    let createdAt: Date?
    let updatedAt: Date?
    let completedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case eventId = "event_id"
        case userId = "user_id"
        case action, generation, status
        case providerEventId = "provider_event_id"
        case expectedProviderRevision = "expected_provider_revision"
        case forceOverwrite = "force_overwrite"
        case attempts
        case maxAttempts = "max_attempts"
        case nextRetryAt = "next_retry_at"
        case failureCode = "failure_code"
        case failureDetail = "failure_detail"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case completedAt = "completed_at"
    }
}
